### PR TITLE
Added an updater.

### PR DESCRIPTION
**Pull request by 4rzael**
# How to use the updater:

* change the PRIVATE.json

* Follow README.md instructions in order to create a dummy SSH user

* restart pheromon

* In the Admin page, select sensors to update, then type `startupdate [PLAYBOOK_FILE_NAME]`
    * In order to stop an update, send `stopupdate` (stops all updates)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -165,6 +166,52 @@ Once built, you can use
...
+sudo passwd sensorSSH;
+```
+
+* Create ssh keys for the user
+
+```sh
+su sensorSSH -c 'ssh-keygen -t rsa -b 4096';
+```
+
+* Add te sensor public key to the authorized_keys and vice-versa
```
Comment (DavidBruant) [inline README.md]: typo: te => the

File: README.md [hunk 2]
```diff
@@ -165,6 +166,52 @@ Once built, you can use
...
+```sh
+sudo usermod -G docker sensorSSH;
+```
+
+* Add a volume with the private key of the user (read only is enough)
+
+  * Idem for the known_hosts
+
+
+* Add theses lines to `/etc/ssh/sshd_config`
```
Comment (DavidBruant) [inline README.md]: In the host? or in the container ?

File: compose-dev.yml [hunk 3]
```diff
@@ -1,20 +1,25 @@
...
       - POSTGRES_USER=postgres
       - POSTGRES_PASSWORD=elements
       - PGPASSWORD=elements
+      - UPDATER_RANGE_START=2200
+      - UPDATER_RANGE_SIZE=10
       - NODE_ENV=development
    volumes:
       - ./:/pheromon
+      - /home/sensorSSH/.ssh/id_rsa:/root/.ssh/id_rsa:ro
+      - /home/sensorSSH/.ssh/known_hosts:/root/.ssh/known_hosts
```
Comment (DavidBruant) [inline compose-dev.yml]: not `:ro` so the sensors can add themselves?

File: api/api.js [hunk 4]
```diff
@@ -37,7 +37,7 @@ schedule.scheduleJob('0 3 * * *', function(){
     var gzip = zlib.createGzip();
     var today = new Date();
     var wstream = fs.createWriteStream('/pheromon/data/backups/' + today.getDay() + '.txt.gz');
-    var proc = spawn('pg_dump', ['-p', process.env.DB_PORT_5432_TCP_PORT, '-h', process.env.DB_PORT_5432_TCP_ADDR, '-U', process.env.POSTGRES_USER, '-d', process.env.POSTGRES_USER, '-w']);
+    var proc = spawn('pg_dump', ['-p', process.env.POSTGRES_PORT, '-h', process.env.POSTGRES_HOST, '-U', process.env.POSTGRES_USER, '-d', process.env.POSTGRES_USER, '-w']);
```
Comment (DavidBruant) [inline api/api.js]: Why this change?

File: api/api.js [hunk 5]
```diff
@@ -99,5 +99,5 @@ server.listen(PORT, function () {
 
 process.on('uncaughtException', function(e){
     console.error('uncaught', e, e.stack);
-    process.kill();
+    process.exit(1);
```
Comment (DavidBruant) [inline api/api.js]: Good catch :-p

File: compose-test.yml [hunk 6]
```diff
@@ -1,16 +1,17 @@
...
-   links:
-      - dbtest:db
-      - broker
-   ports:
-      - "4000:4000"
+   net: host
    environment:
+      - VIRTUAL_HOST=pheromon.ants.builders
       - VIRTUAL_PORT=4000
+      - POSTGRES_HOST=localhost
```
Comment (DavidBruant) [inline compose-test.yml]: Not sure `localhost` is the best value here. Why this choice?

File: compose-prod.yml [hunk 7]
```diff
@@ -1,20 +1,25 @@
 api:
    build: ./
    command: nodemon api/api.js
-   links:
-      - db
-      - broker
-   ports:
-      - "3900:3900"
+   net: host
```
Comment (DavidBruant) [inline compose-prod.yml]: Is that the part where the network interface of the Docker is mapped directly to the host?

File: api/api.js (hunk 4 above)
Comment (DavidBruant) [inline api/api.js]: Per discussion with @4rzael, this was done because `api: net: host` in docker-compose is incompatible with `link` (which makes sense). The solution is to create a new docker service that handles only the reverse SSH business (and revert the api service and code at this line to where it was). To be done in an upcoming commit.

File: compose-prod.yml (hunk 7 above)
Comment (DavidBruant) [inline compose-prod.yml]: See https://github.com/anthill/pheromon/pull/75/files#r45721423

File: compose-dev.yml (hunk 3 above)
Comment (4rzael) [inline compose-dev.yml]: yes

File: api/api.js (hunk 4 above)
Comment (4rzael) [inline api/api.js]: Finally, the `--net=host` can be avoided by using the docker host address (`172.X.0.1`).

We just need to find a way to get this address programmatically.

File: README.md (hunk 2 above)
Comment (4rzael) [inline README.md]: It was in the host, but I don't think we still need it.
